UnitTests/UnitTests.cpp

```diff
@@ -0,0 +1,153 @@
+#include "CppUnitTest.h"
+
+#include "../RTree.h"
+
+#include <array>
+
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
```

Comment:
DSA does all it's developed in visual studio using their unit test suite and therefore any testing I do will need to be done in it.  I do not have resources to port this over into a different test suite.

However, I intentionally added it as separate commit so that you can easily cherry pick the source code without having to accept the unit tests.  If you wish to port the tests into a cpp file as you have done with other tests then feel free.  I included them mainly so you could see the extent the testing that I have performed to date.

As you may have guessed I intend to use this code as part of a larger project and will likely perform more unit testing on this library.  However, as I said before it will need to be in MSVC.

Reply:
Any updates on this guys? NNSearch is a great function indeed, but having the VS dependency in the repo would let my generator Ninja confused. I use VS also but avoid using its .SLN model still.

Reply:
Hi Drew.  My position on this is unchanged.  I do not have resources to port testing.  The options (as I see them are):
1. Testing can be ported over by yourself (or someone else).
2. The method can be cherry picked into the main branch without the tests.

Reply:
I just finished writing a GTest suite for this repo and will incorporate the NN search soon! Thanks for the ping and patience! :)